tests/unit/plugins/test_ant.py

```diff
@@ -122,21 +200,120 @@ def test_env(self):
             env,
         )
 
-    def test_env_proxies(self):
+    def test_build_env_proxies(self):
         env_vars = (
             ("http_proxy", "http://user:pass@localhost:3132"),
             ("https_proxy", "http://user2:pass2@localhost2:3133"),
         )
         for key, value in env_vars:
             self.useFixture(fixtures.EnvironmentVariable(key, value))
-        plugin = ant.AntPlugin("test-part", self.options, self.project_options)
 
-        env = plugin.env(plugin.partdir)
-        self.assertIn(
-            "ANT_OPTS='"
-            "-Dhttp.proxyHost=localhost -Dhttp.proxyPort=3132 "
-            "-Dhttp.proxyUser=user -Dhttp.proxyPassword=pass "
-            "-Dhttps.proxyHost=localhost2 -Dhttps.proxyPort=3133 "
-            "-Dhttps.proxyUser=user2 -Dhttps.proxyPassword=pass2'",
-            env,
+        plugin = ant.AntPlugin("test-part", self.options, self.project)
+
+        env = plugin._build_environment()
+        self.assertThat(env, Contains("ANT_OPTS"))
+        self.assertThat(
+            env["ANT_OPTS"],
+            Equals(
+                "-Dhttp.proxyHost=localhost -Dhttp.proxyPort=3132 "
+                "-Dhttp.proxyUser=user -Dhttp.proxyPassword=pass "
+                "-Dhttps.proxyHost=localhost2 -Dhttps.proxyPort=3133 "
+                "-Dhttps.proxyUser=user2 -Dhttps.proxyPassword=pass2"
+            ),
+        )
+
+
+class AntPluginUnsupportedBase(unit.TestCase):
+    def setUp(self):
```

Comment:
Perhaps it would be better to have a `AntPluginBase` with a `setUp` like this that can be used for all tests in here? Then this one test can just make a `Project` instance instead of duplicating this.

Reply:
it is a different project options though